review/static/js/src/review.js

```diff
@@ -0,0 +1,75 @@
+/* Javascript for ReviewXBlock. */
+function ReviewXBlock(runtime, element) {
+  $(function ($) {
+    'use strict'
+
+    $('.review-button').on('click', function(event) {
+      var $btn = $(event.currentTarget),
+          $content = $btn.siblings('.review-content'),
+          $iframe = $content.find('iframe'),
+          title,
+          skipLink,
+          parser;
+
+      if (!$btn.hasClass('disable-click')) {
+        /*
+        There is an issue where the buttons containing the review content
+        could have multiple listeners on it causing it to remain closed when
+        it was clicked on since the first listener would open it and the
+        subsequent listener would immediately close it. This disables clicking
+        for a short time so additional listeners can't happen
+        Use case: In studio, every time you open the edit window, it attaches
+        a new listener and then the above issue occurs.
+        */
+        $btn.addClass('disable-click');
+
+        setTimeout(function() {
+          $btn.removeClass('disable-click');
+        }, 100);
+
+        // iFrame loads after the button is clicked so there is not
+        // a large amount of loading upon going to the xBlock
+        if ($iframe.attr('src') === '') {
+          $iframe.attr('src', $btn.data('iframe-src'));
+        }
+
+
+        // Prevent issues with cross-origin frames when in studio
+        // (iFrames are hosted on courses.edx.org)
+        parser = document.createElement('a');
+        parser.setAttribute('id', 'iframe-hostname');
+        parser.href = $btn.data('iframe-src');
+        if (parser.hostname == window.location.hostname) {
```

Comment:
@AlasdairSwan Can I get your feedback on this method of preventing Cross-Origin Request issues? The old way was hard-coding 'studio' instead of parser.hostname

Reply:
Seems fine @Dillon-Dumesnil, although it is preferable to use ```===``` over ```==```